apps/src/turtle/artist.js

```diff
@@ -676,15 +700,43 @@ Artist.prototype.reset = function (ignore) {
  * Click the run button.  Start the program.
  */
 Artist.prototype.runButtonClick = function () {
+  this.shouldAnimate_ = !this.instant_;
   this.studioApp_.toggleRunReset('reset');
   document.getElementById('spinner').style.visibility = 'visible';
   if (this.studioApp_.isUsingBlockly()) {
     Blockly.mainBlockSpace.traceOn(true);
   }
   this.studioApp_.attempts++;
+  if (this.limitedAutoRun) {
+    this.immovableBlocks = [];
+    Blockly.mainBlockSpace.getAllBlocks().forEach(block => {
+      if (!block.isMovable()) {
+        this.immovableBlocks.push(block);
+      } else {
+        block.setMovable(false);
```

Comment:
is this gonna mess up any blocks that are supposed to be immovable on initialization?

Reply:
It shouldn't because I'm tracking the blocks that were initially immovable in `this.immovableBlocks`. However, it can cause problems if the project autosaves when you're in the immovable mode and you reload the page. I changed the implementation by adding `lockMovement()`/`unlockMovement()` methods to blockly in https://github.com/code-dot-org/blockly/pull/145.